src/common/finite_element_state.hpp

```diff
@@ -65,7 +65,7 @@ class FiniteElementState {
    * the dimension of the FESpace, the type of FEColl, the DOF ordering that should be used,
    * and the name of the field
    */
-  FiniteElementState(std::shared_ptr<mfem::ParMesh> pmesh, FEStateOptions&& options = FEStateOptions());
+  FiniteElementState(mfem::ParMesh& pmesh, FEStateOptions&& options = FEStateOptions());
```

Comment:
Should we be labeling in parameters as non-owning?

Reply:
Are references ever owning? I always assume references are owned by the caller.

Reply:
You could explicitly delete it but that would be bad. I was talking more for about consistency and clarity sake. We do return it out of this class below and label it as non-owning... how do you feel about this @joshessman-llnl ?

Reply:
I don't mind adding it in for clarity, but Jamie is right that references are almost never owning  - you'd have to do something really weird to get an "owning reference":
```cpp
int& a = *(new int);
// ...
delete &a;
```

Reply:
I think a note saying not to do this in the style guide would suffice.